### DISCUSSION TITLE
`buildFromSdist` gets set to false when `defaults.settings.default` is present

### DISCUSSION THREAD

**Discussion by johnhampton** [srid/haskell-flake#301] [Q&A]
#298 ensures that `buildFromSdist` is set to true for local packages. However if `defaults.settings.default.haddock` is set, `buildFromSdist` gets reset to `false`.

This results in a build with `buildFromSdist` set to `true`.
```nix
        haskellProjects.default = {
          # defaults.settings.default = { haddock = true;  };

          # Project settings here
        };
```
This results in a build with `buildFromSdist` set to `false`.
```nix
        haskellProjects.default = {
          defaults.settings.default = { haddock = true;  };

          # Project settings here
        };
```

I can force it back to true with:
```nix
        haskellProjects.default = {
          defaults.settings.default = { haddock = true; buildFromSdist = true; };

          # Project settings here
        };
```

**Comment by srid** [comment]
This is expected behaviour, @johnhampton ... but see #275 which may be useful to your use-case.

**Reply by johnhampton** [reply]
Interesting, I wouldn't expect setting `defaults.settings.default.haddock` to change the value of `defaults.settings.default.buildFromSdist`. Is that documented somewhere? If not, maybe consider documenting.

**Reply by srid** [reply]
Because, `defaults.settings.default.haddock` is not an option; `defaults.settings.default` is. When you set `defaults.settings.default.haddock = true;` you are in effect setting `defaults.settings.default = { haddock = true; }` which then overrides the option's default value (an attrset).

We can document this option as part of https://github.com/srid/haskell-flake/pull/275
